From e67b7f8d7a719db34a71cbcd793237ccaf6cf6d5 Mon Sep 17 00:00:00 2001
From: Simon Arlott <sa.me.uk>
Date: Mon, 19 May 2025 18:46:50 +0100
Subject: [PATCH 1/2] RP2040: us_ticker: use API to force a timer event and get
 an interrupt

us_ticker_irq_handler() must only be called from interrupt context
---
 targets/TARGET_RASPBERRYPI/TARGET_RP2040/us_ticker.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/targets/TARGET_RASPBERRYPI/TARGET_RP2040/us_ticker.c b/targets/TARGET_RASPBERRYPI/TARGET_RP2040/us_ticker.c
index b3b8497188c..3a5e1f1686f 100644
--- a/targets/TARGET_RASPBERRYPI/TARGET_RP2040/us_ticker.c
+++ b/targets/TARGET_RASPBERRYPI/TARGET_RP2040/us_ticker.c
@@ -99,7 +99,7 @@ void us_ticker_set_interrupt(timestamp_t timestamp)
 
 void us_ticker_fire_interrupt(void)
 {
-    us_ticker_irq_handler();
+    hardware_alarm_force_irq(alarm_num);
 }
 
 void us_ticker_disable_interrupt(void)
